benchmark.go

```diff
@@ -188,12 +296,82 @@ func runBenchmark(out output, logger *zap.Logger, allOpts Options, resolved reso
 		zap.Time("startTime", start),
 	)
 
+	// Print out errors
+	// TODO: allow errors to be printed in JSON format for output consistency
 	overall.printErrors(out)
-	overall.printLatencies(out)
 
-	out.Printf("Elapsed time:      %v\n", (total / time.Millisecond * time.Millisecond))
-	out.Printf("Total requests:    %v\n", overall.totalRequests)
-	out.Printf("RPS:               %.2f\n", float64(overall.totalRequests)/total.Seconds())
+	latencyValues := overall.getLatencies()
+
+	// Rounding RPS value to the hundredths place
+	rps := float64(overall.totalRequests) / total.Seconds()
+	rps = (math.Round(rps * 100)) / 100
+
+	summary := Summary{
+		ElapsedTime:   (total / time.Millisecond * time.Millisecond).String(),
+		TotalRequests: overall.totalRequests,
+		RPS:           rps,
+	}
+
+	if formatAsJSON {
+		outputJSON(out, benchmarkParameters, latencyValues, summary)
+	} else {
+		outputPlaintext(out, latencyValues, summary)
+	}
+}
+
+// JSON output helper method
```

Comment:
nit: this comment doesn't provide much additional context over the method name (which is pretty self-describing), so I think it can be removed. same below for the plain text